### DISCUSSION TITLE
Allowe duration based entries

### DISCUSSION THREAD

**Discussion by 3Fish** [ivan-lednev/obsidian-day-planner#349] [Features & bugs]
I think often you have elements where you're roughly how long they will take, but not when you will be able to do them. 
I would like to have the day planer consider the order of tasks and just insert a block of the given duration after the previous task. If the previous task has an end time, it should be directly after it; otherwise it should be moved as the time progresses, but stops moving back when it collides with the next fixed task. Multiple duration tasks would then just move together.
If a previous open end task is marked as done, the completion time is used as the end date for the purpose of moving the duration-based tasks.

The format could be something like ~15 for minutes and ~2:30 for hours and minutes.
```
- [ ] 09:15 - 10:00 Read emails
- [ ] ~15 Get a Coffee
- [ ] 12:00 Meeting with Customer
- [ ] ~1:00 Get lunch
- [ ] 17:00 Pick up the Kids
```
it is before 12 o'clock
![it is before 12 o'clock](https://github.com/ivan-lednev/obsidian-day-planner/assets/4740439/1a890321-a3d2-401a-bda7-d4913f77b577)
it is 13 o'clock
![it is 13 o'clock](https://github.com/ivan-lednev/obsidian-day-planner/assets/4740439/9b5b72f5-ba6b-4b7d-b530-ce6756483bd3)
any time after 16 o'clock
![any time after 16 o'clock](https://github.com/ivan-lednev/obsidian-day-planner/assets/4740439/6f2d39c3-694a-4acb-8a41-f689a48662b6)
the customer meeting was done at 14 o'clock
![the customer meeting was done at 14 o'clock](https://github.com/ivan-lednev/obsidian-day-planner/assets/4740439/b7d7f49c-8215-403e-b66f-372850cef26b)

**Comment by GroveTomas** [comment]
Totally support this feature request!
I found the day planner just because I bumped into this plugin for roam research [Nautilus](https://github.com/tombarys/roam-depot-nautilus/tree/6-7-8?tab=readme-ov-file) in one of my newsletters.
Love the idea. But I have no idea how hard it is to implement.
